### DISCUSSION TITLE
source command not found in .sh files

### DISCUSSION THREAD

**Discussion by arojas314** [MAAP-Project/Community#787] [Platform] [answered]
Hi all,
I am getting error `source: not found` when running `source activate my-custom-package` or `source activate base` in any .sh file. I receive this error using the Basic Stable jupyter workspace. Container: [mas.maap-project.org/root/maap-workspaces/jupyterlab3/vanilla:v3.0.1](http://mas.maap-project.org/root/maap-workspaces/jupyterlab3/vanilla:v3.0.1).
The error actually comes up for any `.sh` file that uses the `source` command to activate environments, not just when I run a DPS job. 
Any suggestions?
Thanks in advance!

**Comment by wildintellect** [comment]
@sujen1412 did anything change in the new ADE? source was working in the old  ADE/DPS https://github.com/lauraduncanson/icesat2_boreal/blob/master/dps/alg_3-4/run_boreal_biomass.sh

@arojas314  please provide a full example where you setup a custom conda env and then try to activate it with source

In the ADE you might need to use `conda activate` not `source` but it was working in DPS previously

**Reply by arojas314** [reply]
This happens using `source activate base` in any  `.sh` file.

**Reply by arojas314** [reply]
@wildintellect using `conda activate base` i receive the following error: `CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.`

**Reply by arojas314** [reply]
this is my `.sh` file:
```
#!/bin/bash

# conda activate base
source activate base
```

**Reply by wildintellect** [reply]
ADE - jupyter used `conda activate`
DPS - likely uses `source activate`
but you shouldn't need to activate the `base` in DPS

**Reply by arojas314** [reply]
Thanks @wildintellect. Those commands do not work in my `.sh` files. So I am unable to activate any environment (base or custom).

**Reply by arojas314** [reply, marked as answer]
I wanted to update answer for this. I was calling file to test using `sh run.sh`, but the correct way would be to use bash i.e. `bash run.sh`
